pyproject.toml

```diff
@@ -84,8 +84,7 @@ dependencies = [
   "mlflow",
 
   # Elasticsearch
-  "elasticsearch>=7.7,<7.11",
-  "elastic-apm",
+  "elasticsearch>=7.16.2,<8",
```

Comment:
This change forces every current user of Haystack and Elasticsearch to upgrade the dependency. Before this change they had <7.11 and now they need to have >=7.16.2. Would it also work with "elasticsearch>=7.7,<8" instead? If not, we should tag this PR as a breaking change and explain it in the next release notes.

Reply:
The fix should work with 7.7 onwards so I think it only depends on how strong we feel about running Haystack with Elasticsearch versions that are vulnerable to the log4j bug. Probably fine to avoid a breaking change and invite our users to upgrade their Elasticsearch instances (e.g. in the release notes)